### PR TITLE
API Capabilities

### PR DESCRIPTION
**Pull request by leplatrem**
I suggest we expose something like this on the Python side for plugins:

```python
config.add_api_capability("identifier", description="", documentation="")
```
> It would raise an error if already defined for example

And shown in the root URL as a mapping:

```js
"capabilities": {
  "auth-fxa": {
    "description": "Firefox Account authentication",
    "documentation": "http://github.com/mozilla-services/cliquet-fxa"
  }
}
```

@almet @n1k0 @Natim thoughts?

### CODE REVIEW

**Comment by Natim** [comment]
I totally like the idea :)

**Comment by leplatrem** [comment]
This is now open for review/merge :)

File: cliquet/__init__.py [hunk 1]
```diff
@@ -117,6 +117,18 @@ def includeme(config):
     # Public settings registry.
     config.registry.public_settings = {'batch_max_requests', 'readonly'}
 
+    # Directive to declare arbitrary API capabilities.
+    def add_api_capability(config, identifier, description="", url="", **kw):
+        if identifier in config.registry.api_capabilities:
+            error_msg = "API capability '%s' already exists." % identifier
```
Comment (almet) [inline cliquet/__init__.py]: I think we can be more explicit here: `"The '%s' capability is already defined, with the values %s. Only one plugin can provide one capability" % (identifier, config.registry.api_capabilites[identifier])`.

File: cliquet_docs/api/utilities.rst [hunk 2]
```diff
@@ -23,6 +23,10 @@ The returned value is a JSON mapping containing:
   - ``batch_max_requests``: Number of requests that can be made in a batch request.
   - ``readonly``: Only requests with read operations are allowed.
 
+- ``capabilities``: a mapping used by clients to detect optional features of the API.
+
+  - Example: ``{"auth-fxa": {"description": "Firefox Account authentication", "url": "http://github.com/mozilla-services/cliquet-fxa"}}``
```
Comment (almet) [inline cliquet_docs/api/utilities.rst]: The example would be better if the json is expended.

File: cliquet_docs/quickstart.rst [hunk 3]
```diff
@@ -200,6 +200,33 @@ Advanced initialization
 .. autofunction:: cliquet.initialize
 
 
+
+Declare API capabilities
```
Comment (almet) [inline cliquet_docs/quickstart.rst]: I would put this in the "plugins" section of the docs and not in the quickstart.

**Comment by almet** [comment]
r+wc

**Comment by michielbdejong** [comment]
Cool stuff!

**Comment by Natim** [comment]
Yes great :+1:

**Comment by leplatrem** [comment]
Thanks for your positive feedback!

**Pull request merged**